src/plasma-cash/user.ts

```diff
@@ -83,46 +82,48 @@ export class User extends Entity {
 
   async depositETHAsync(amount: BN): Promise<IPlasmaCoin> {
     let currentBlock = await this.getCurrentBlockAsync()
-    const tx = await this.sendETH(this.plasmaCashContract._address, amount, 220000)
-    const coin = await this.getCoinFromTxAsync(tx, 0)
+    const tx = await this.web3.eth.sendTransaction({
+      from: this.ethAddress,
+      to: this.plasmaCashContract._address,
+      value: this.web3.utils.toHex(amount),
+      gas: this._defaultGas
+    })
+    const coin = await this.getCoinFromTxAsync(tx.transactionHash, 0)
     currentBlock = await this.pollForBlockChange(currentBlock, 20, 2000)
     this.receiveAndWatchCoinAsync(coin.slot)
     return coin
   }
 
   async depositERC721Async(uid: BN, address: string): Promise<IPlasmaCoin> {
-    // @ts-ignore
-    const token = new SignedContract(this._web3, ERC721, address, this.ethAccount).instance
+    const token = new this.web3.eth.Contract(ERC721, address)
     let currentBlock = await this.getCurrentBlockAsync()
-    const tx = await token.safeTransferFrom([
-      this.ethAccount.address,
-      this.plasmaCashContract._address,
-      uid.toString()
-    ])
-    const coin = await this.getCoinFromTxAsync(tx, 1) // 2 events, transferred & deposited, we want the 2nd one
+    const tx = await token.methods
+      .safeTransferFrom(this.ethAddress, this.plasmaCashContract._address, uid.toString())
+      .send({ from: this.ethAddress, gas: this._defaultGas })
+    const coin = await this.getCoinFromTxAsync(tx.transactionHash, 1) // 2 events, transferred & deposited, we want the 2nd one
     currentBlock = await this.pollForBlockChange(currentBlock, 20, 2000)
```

Comment:
It keeps blocked here, when call the function. Probably because ganache only produces blocks for transactions? not sure

Reply:
I'm not sure what you mean. This should work, given that the tests in https://github.com/loomnetwork/plasma-cash/tree/metamask-update use that function.